### PR TITLE
Task/hw 61330 iOS update ui SDK to send notification for authentication error

### PR DESCRIPTION
**Pull request by asenthil-altmtrk**

### CODE REVIEW

File: Common/Sources/View/Error/ErrorView.swift [hunk 1]
```diff
@@ -57,7 +57,11 @@ public final class ErrorView {
             connectionError({ (_) in handler?() })
 
         default:
-            unexpectedError()
+            if error.getAuthenticationError() != nil {
```
Comment (manikandan-altmtrk) [inline Common/Sources/View/Error/ErrorView.swift]: FYI: In android we worked like this.
Screenshot 2020-03-19 at 11 46 44 AM
Screenshot 2020-03-19 at 11 47 09 AM

Comment (asenthil-altmtrk) [reply Common/Sources/View/Error/ErrorView.swift]: But in ios that was the requirement

Comment (jsaini-hw) [reply Common/Sources/View/Error/ErrorView.swift]: Good point Mani. We can actually do similar thing in iOS. @asenthil-altmtrk instead of checking for error.getAuthenticationError() != nil in UI SDK and iOS App, update `public var group: HyperwalletErrorGroup ` in HyperwalletError.swift (Hyperwallet Core SDK) to add another case for authenticationError from 
```suggestion
            case .authenticationError:
                return HyperwalletErrorGroup.authenticationError
```
and remove it from unexpected case.

Comment (jsaini-hw) [reply Common/Sources/View/Error/ErrorView.swift]: Also then update ErrorView in HyperwalletApp `case .unexpected:
            if error.getAuthenticationError() != nil {
                authenticationError(title, handler)
            } else {
                unexpectedError({ (_) in handler?() })
            }
        }` to check for .authenticationError instead.

Comment (asenthil-altmtrk) [reply Common/Sources/View/Error/ErrorView.swift]: I have updated in Core SDK, UISDK  and App

File: Cartfile [hunk 2]
```diff
@@ -1,2 +1,2 @@
-github "hyperwallet/hyperwallet-ios-sdk" "development"
+github "hyperwallet/hyperwallet-ios-sdk" "task/HW-61330-iOS-Update-UI-SDK-to-send-notification-for-authentication-error"
```
Comment (jsaini-hw) [inline Cartfile]: I have approved Core SDK PR, coordinate with QA to get that merged 1st and then revert this change before merging this one

Comment (asenthil-altmtrk) [reply Cartfile]: I have updated the cart file

File: Common/Sources/Extensions/Notification.swift [hunk 3]
```diff
@@ -53,4 +58,6 @@ public enum UserInfo: String {
     case transferCreated
     /// A transfer has been scheduled.
     case transferScheduled
+    ///  Authentication error has been occured
```
Comment (nirmal-altmtrk) [inline Common/Sources/Extensions/Notification.swift]: Remove extra space after ///.

File: Common/Sources/View/Error/ErrorView.swift [hunk 4]
```diff
@@ -111,4 +114,18 @@ public final class ErrorView {
...
                                                 handler)
     }
+
+    private func authenticationError(_ error: HyperwalletErrorType) {
+        HyperwalletUtilViews.showAlert(viewController,
+                                       title: "authentication_error_title".localized(),
+                                       message: error.getAuthenticationError()?.message() ??
+                                        "authentication_error_message".localized(),
+                                       actions: UIAlertAction.close({ (_) in
+                                        NotificationCenter.default
```
Comment (nirmal-altmtrk) [inline Common/Sources/View/Error/ErrorView.swift]: Update the indent as below.
Screenshot 2020-03-23 at 3 59 44 PM

Please reindent the whole class.

File: UITests/TransferFunds/TransferUserFundsTest.swift [hunk 5]
```diff
@@ -790,7 +790,7 @@ class TransferUserFundsTest: BaseTests {
             || destinationDetail == "United States Ending on 6789")
 
         transferFunds.addSelectDestinationLabel.tap()
-
+        waitForNonExistence(spinner)
```
Comment (nirmal-altmtrk) [inline UITests/TransferFunds/TransferUserFundsTest.swift]: Add new line after this line.

Comment (asenthil-altmtrk) [reply UITests/TransferFunds/TransferUserFundsTest.swift]: Added new line

File: Common/Sources/View/Error/ErrorView.swift (hunk 4 above)
Comment (asenthil-altmtrk) [reply Common/Sources/View/Error/ErrorView.swift]: Indent added

File: Common/Sources/Extensions/Notification.swift (hunk 3 above)
Comment (asenthil-altmtrk) [reply Common/Sources/Extensions/Notification.swift]: Removed extra space

File: Common/Sources/View/Error/ErrorView.swift [hunk 6]
```diff
@@ -111,4 +114,16 @@ public final class ErrorView {
                                                 message: "network_connection_error_message".localized(),
                                                 handler)
     }
+
+    private func authenticationError(_ error: HyperwalletErrorType) {
+        HyperwalletUtilViews.showAlert(viewController,
```
Comment (jsaini-hw) [inline Common/Sources/View/Error/ErrorView.swift]: Add HyperwalletInsights.shared.trackError

Comment (asenthil-altmtrk) [reply Common/Sources/View/Error/ErrorView.swift]: Added track error with an Exception type error
